docs/developer/extending/apps/developing-with-tunnels.mdx

```diff
@@ -0,0 +1,106 @@
+
+As you want Apps to react to events in your store, you reach for webhooks. However, webhooks pose a unique challenge: they can only communicate with public clients. If so, how are you supposed to develop your app on `localhost`?
+
+This is where tunnels come into play. **A tunnel allows you to create a proxy that exposes a local environment to a public URL.**
+
+```mermaid
+flowchart LR
+    a[Dev server]
+    t[Tunnel]
+    c[Saleor Cloud]
+    e[External services]
+
+    subgraph Local environment
+        a
+    end
+
+    subgraph Internet
+        t
+        c
+        e
+    end
+
+    a<-.-> t
+    t<-->c
+    t<-->e
+```
+
+---
+
+Let's say you are building a [Next.js](https://nextjs.org/) application that leverages a webhook. You boot it up with the `next dev` command, which renders it on `http://localhost:3000`. Since this address is not HTTPS-compliant, the app cannot access the webhook response. We have a perfect use case for a tunnel on our hands.
+
+A simplified process of using a tunnel to expose this app would look like this:
+
+1. You authenticate yourself in the tunneling service of your choosing (e.g.: `ngrok`).
+2. You provide the tunnel with the port number `3000` (where your app is running).
+3. You access the URL returned from the tunnel.
+4. You successfully access the response from the webhook 🎉.
+
+## Available tunnels
+
+### Saleor App Tunnel
+
+:::caution
+We are keen on making Saleor App Tunnel flawless. Yet, issues with its stability may occur now and then. If that's the case, please look at [the alternatives](#alternatives).
+:::
+
+To use Saleor App Tunnel, you must install [Saleor CLI](https://www.npmjs.com/package/@saleor/cli).
+
+:::note
+Run `saleor app tunnel -h` to check detailed usage.
+:::
+
+To create a tunnel, run `saleor app tunnel`. After several seconds, the CLI should print a URL with the`.saleor.live` domain:
+
+```bash
+$ saleor app tunnel
+
+✔ Organization · YourShop
+✔ Select Environment · staging
+✔ Environment · staging
+─────────────────────────────────────────────────────────────────
+
+     Saleor App Name: Your app
+     Saleor App URL: https://your-app-shop.saleor.live
+
+     Saleor Dashboard: https://yourshop.eu.saleor.cloud/dashboard/
+     GraphQL Playground: https://yourshop.eu.saleor.cloud/graphql/
+
+─────────────────────────────────────────────────────────────────
+✔ Installing...
+
+Tunnel is listening to your local machine on port: 3000
+
+Press CTRL-C to stop the tunnel
+Use it to install the app in the Dashboard
+
+```
+
+#### Installing the App in the Dashboard
+
+An extra feature of the Saleor App Template is the ability to install Apps inside the Dashboard. This way, you can develop the App while displaying it inside your Saleor Dashboard.
+
+To do it, please run the following:
+
+```
+saleor app tunnel --force-install
+```
+
```

Comment:
Maybe it's worth mentioning, that theres `saleor app install` command which can be used in case tunnel is already up (both our or alternative like ngrok)?
The command has to be also run from the app directory